src/index.mli

```diff
@@ -158,3 +139,29 @@ end
 
 module Make (K : Key) (V : Value) (IO : IO) :
   S with type key = K.t and type value = V.t
+
+(** These modules should not be used. They are exposed purely for testing
+    purposes. *)
+module Private : sig
+  module Hook : sig
+    type 'a t
+
+    val v : ('a -> unit) -> 'a t
+  end
+
+  module Search : module type of Search
+
+  module Io_array : module type of Io_array
+
+  module Fan : module type of Fan
+
+  module type S = sig
+    include S
+
+    val force_merge : ?hook:[ `After | `Before ] Hook.t -> t -> unit
+    (** [force_merge t] forces a merge for [t]. *)
```

Comment:
Not strictly related to this PR, but we could take advantage of it to specify a bit more clearly how to use the hooks.

Reply:
Indeed: done in https://github.com/mirage/index/pull/147/commits/6e1395c0de735414f56cd0e3b21a3e8687fef477.

Reply:
Looks good, maybe just specify that it happens in the child thread?